packages/cli/src/databases/repositories/workflow.repository.ts

```diff
@@ -1,10 +1,71 @@
 import { Service } from 'typedi';
 import { DataSource, Repository } from 'typeorm';
 import { WorkflowEntity } from '../entities/WorkflowEntity';
+import { mixinQueryMethods } from './mixins/queryMethods.mixin';
+import {
+	validateSync,
+	IsOptional,
+	IsString,
+	IsBoolean,
+	IsArray,
+	IsDateString,
+} from 'class-validator';
+import { Schema } from './schema';
+
+export class WorkflowSchema extends Schema {
+	constructor(data: unknown = {}) {
+		super();
+		Object.assign(this, data);
+
+		// strip out unknown fields
+		const result = validateSync(this, { whitelist: true });
+
+		if (result.length > 0) {
+			throw new Error('Parsed filter does not fit the schema');
+		}
+	}
+
+	@IsOptional()
+	@IsString()
+	id?: string = undefined;
+
+	@IsOptional()
+	@IsString()
+	name?: string = undefined;
+
+	@IsOptional()
+	@IsBoolean()
+	active?: boolean = undefined;
+
+	@IsOptional()
+	@IsArray()
+	@IsString({ each: true })
+	nodes?: string[] = undefined;
+
+	@IsOptional()
+	@IsDateString()
+	createdAt?: Date = undefined;
+
+	@IsOptional()
+	@IsDateString()
+	updatedAt?: Date = undefined;
```

Comment:
I'm wondering if we could somehow generate this from the existing db entity 🤔

Reply:
Agree about the duplication, also not happy about having to explicitly assign `undefined`. 

Next up is the credentials list query - let's revisit this then.

Reply:
Discussed with @krynble - we could create a `@Filterable` decorator and derive the filter schema from it, but let's not block on this.